### ISSUE TITLE
Feature Request: Add ability for ToggleButton to operate like an HTML radio button

### ISSUE THREAD

**Issue by davidklebanoff** [appbaseio/reactivesearch#286]
Issue Type:
Enhancement

Description:
Right now the ToggleButton component acts like a group of HTML checkboxes, which is fine, but there is room for more flexibility. It would be useful to add the ability to let the ToggleButton component operate like an HTML radio button; meaning, it can be configured such that one value must be selected at all times.

Additionally, if the feature is added, it would be useful to have the ability to add an option that when selected is the equivalent to a no-op, meaning it would operate the same if no options were selected. Maybe something like:

 ```jsx
<ToggleButton
  componentId="MeetupTops"
  dataField="group_topics.topic_name.raw"
  data={
    [{"label": "All"},
     {"label": "Social",   "value": "Social"},
     {"label": "Travel",   "value": "Travel"},
     {"label": "Outdoors", "value": "Outdoors"}]
  }
  multiSelect={false}
  defaultSelected=["All"]
/>
```

This way, an option is always clearly selected, but you still have the flexibility of searching across all categories.

**Comment by davidklebanoff** [comment]
Optionally, instead of making the ToggleButton component flexible enough to act both like a checkbox and radio button control, it may make more sense to have two separate components. I'm not sure where you'd want to draw the line.

**Comment by siddharthlatest** [comment]
I agree with the idea. Perhaps a prop called `setRadioMode` or just `radioMode` (defaults to false) will make one of the options as always selectable. When set to this mode, the first label gets selected if no `defaultSelected` is provided. (This will also require some consideration wrt `SelectedFilters` as a radio-mode has no way to be cleared.)  Thoughts on this @metagrover @divyanshu013?

The second part for no-op also makes sense. We have the `selectAllLabel` in SingleList / MultiList components. We can have a similar `selectNoneLabel` here which when set shows the no-op option with the user provided label.

I would ideally want this to be two components but for breaking compatibility of all existing implementations using it.

**Comment by siddharthlatest** [comment]
On a second thought, I think that not showing the component within the `SelectedFilters` shouldn't be a big deal. And there is a valid use-case where you may want to clear the filter even in radioMode.

Another interesting thing about radio mode is that it only applies when `multiSelect` is set to false. Given that, does it make sense to do one of the two:
1. Instead of having a separate radioMode, make `multiSelect={false}` behave as the radio mode. i.e. un-selection of active button isn't possible.
2. Keep both props supported (which we can if there are proportional use-cases), but it comes with cost of remembering / learning to set two props to get the desired behavior. And some selections of the two being incompatible with each other.

I vote for 1. -> an un-selection (clearing) behavior here can be supported via SelectedFilters. We only need to change the UI behavior and keep the `multiSelect` prop to switch between the radio and checkbox modes.
